src/main/java/org/fundacion/jala/converter/view/Controller/ClientRequest.java

```diff
@@ -0,0 +1,158 @@
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.entity.mime.MultipartEntityBuilder;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+import org.fundacion.jala.converter.view.Models.IrequestForm;
+import org.fundacion.jala.converter.view.Models.Parameter;
+import java.io.*;
+
+public class ClientRequest {
+    private String sURL;
+    private CloseableHttpClient httpClient;
+    private HttpPost httpPost;
+    private MultipartEntityBuilder builder;
+    private HttpEntity multipart;
+    private IrequestForm requestForm;
+
+    /**
+     * Http client creates a request given a requestForm.
+     */
+    public ClientRequest(IrequestForm requestForm) {
+        this.requestForm = requestForm;
+        this.sURL = requestForm.getURL();
+        this.httpClient = HttpClients.createDefault();
+    }
+
+    /**
+     * Executes a request given the type of requestForm.
+     * @return
+     * @throws ClientProtocolException
+     * @throws IOException
+     */
+    public String executeRequest(IrequestForm requestForm) throws ClientProtocolException, IOException {
+        String token = authGetToken();
+        httpPost = new HttpPost(sURL);
+        builder = MultipartEntityBuilder.create();
+        multipart = builder.build();
+        addBodyFields();
+        httpPost.setEntity(multipart);
+        httpPost.setHeader("Authorization", "Bearer " + token);
+        CloseableHttpResponse response = httpClient.execute(httpPost);
+        HttpEntity responseEntity = response.getEntity();
+        String sResponse = EntityUtils.toString(responseEntity, "UTF-8");
+        System.out.println("Post return result" + sResponse);
+        return sResponse;
+    }
+
+    /**
+     * Downloads a file from endpoint to given path
+     * @throws ClientProtocolException
+     * @throws IOException
+     */
+    public void download (String filePath) throws IOException{
+        String token = authGetToken();
+        String sURL = "http://localhost:8080/api/download/img1.png";
+
+        CloseableHttpClient httpClient = HttpClients.createDefault();
+        HttpGet request = new HttpGet(sURL);
+        request.setHeader("Authorization","Bearer " + token);
+        CloseableHttpResponse response = httpClient.execute(request);
+        HttpEntity responseEntity = response.getEntity();
+        InputStream inputStream = responseEntity.getContent();
+        FileOutputStream outputStream = new FileOutputStream(new File(filePath));
+        int inByte;
+        while((inByte = inputStream.read()) != -1)
+            outputStream.write(inByte);
+        inputStream.close();
+        outputStream.close();
+    }
+    /**
+     * Retrieves a  token from the endpoint from a username and password.
+     * @return

```

Comment:
All methods should provide a description of their params and returns on their javadocs, not only the name.